lib/ansible/modules/monitoring/zabbix/zabbix_host_triggers.py

```diff
@@ -0,0 +1,254 @@
+                default='hostname',
+                type='str',
+                choices=['hostname', 'visible_name', 'hostid']),
+            trigger_severity=dict(
+                type='str',
+                required=False,
+                default='average',
+                choices=['not_classified', 'information', 'warning', 'average', 'high', 'disaster']),
+            validate_certs=dict(type='bool', required=False, default=True),
+            timeout=dict(type='int', default=10),
+
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg="Missing required zabbix-api module (check docs or install with: pip install zabbix-api)")
+
+    trigger_severity_map = {'not_classified': 0, 'information': 1, 'warning': 2, 'average': 3, 'high': 4, 'disaster': 5}
+    server_url = module.params['server_url']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    http_login_user = module.params['http_login_user']
+    http_login_password = module.params['http_login_password']
+    validate_certs = module.params['validate_certs']
+    host_id = module.params['host_identifier']
+    host_id_type = module.params['host_id_type']
+    trigger_severity = trigger_severity_map[module.params['trigger_severity']]
+    timeout = module.params['timeout']
+
+    host_inventory = 'hostid'
+    zbx = None
+    # login to zabbix
+    try:
+        zbx = ZabbixAPI(server_url, timeout=timeout, user=http_login_user, passwd=http_login_password,
+                        validate_certs=validate_certs)
+        zbx.login(login_user, login_password)
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Zabbix server: %s" % e)
+
+    host = Host(module, zbx)
+
+    if host_id_type == 'hostname':
+        zabbix_host = host.get_host(host_id, host_inventory, 'host')
+        host_id = zabbix_host['hostid']
+
+    elif host_id_type == 'visible_name':
+        zabbix_host = host.get_host(host_id, host_inventory, 'name')
+        host_id = zabbix_host['hostid']
+
+    elif host_id_type == 'hostid':
+        ''' check hostid exist'''
+        zabbix_host = host.get_host(host_id, host_inventory, 'hostid')
+
+    else:
+        module.exit_json(ok=False, host=[], result="No Host present")
+
+    triggers = host.get_triggers_by_host_id_in_problem_state(host_id, trigger_severity)
+
+    triggers_ok = []
+    triggers_problem = []
+    for trigger in triggers:
+        # tGet last event for trigger with problem value = 1
+        # https://www.zabbix.com/documentation/3.4/manual/api/reference/trigger/object
+        if int(trigger['value']) == 1:
+            event = host.get_last_event_by_trigger_id(trigger['triggerid'])
+            trigger['last_event'] = event
+            triggers_problem.append(trigger)
+        else:
+            triggers_ok.append(trigger)
+
+    module.exit_json(ok=True, triggers_ok=triggers_ok, triggers_problem=triggers_problem)
```

Comment:
I think `changed` is what you need here instead of `ok`. However as this module is a "_facts" module and it should never change anything on the remote machine, the value of `changed` should always be `False` (which i believe is the default behavior).

Reply:
@rubentsirunyan the "ok" keyword is like the description field "value" of trigger object (https://www.zabbix.com/documentation/3.0/manual/api/reference/trigger/object).